controllers/graphController.js

```diff
@@ -153,29 +172,123 @@ GraphController.prototype.publicRankedIndex = function(req, res, next) {
 	.catch(next)
 }
 
-// GET /fthr
-GraphController.prototype.userIndex = function(req, res, next) {
-	var username = req.params.model
+// GET /fthr?public=1|0
+GraphController.prototype._userOwnIndex = function(user, req, res, next) {
+	var that = this
+	var username = user.username
+	// the user wants to see only their private graphs
+	var wantPrivate = undefined
+	if (typeof req.query.public !== 'undefined') {
+		wantPrivate = (req.query.public === '0')
+	}
+
+	var makeList = function(list) {
+		if (!list || !list.length)
+			return
+		return list.map(function(graph) {
+			return prettyPrintGraphInfo(graph.toJSON())
+		})
+	}
+
+	function render(publicList, privateList, profile, data) {
+		data = _.extend({
+			bodyclass: '',
+			publicHasMoreLink: false,
+			privateHasMoreLink: false,
+			isSummaryPage: false,
+			withProjectListFilter : true,
+			meta : {
+				header: 'srv/userpage/userpageHeader',
+				title: 'Your Files',
+				scripts: [
+					helper.metaScript('site/userpages.js')
+				]
+			}
+		}, data)
+
+		// format
+		data.publicGraphs = publicList || []
+		data.privateGraphs = privateList || []
+		data.profile = profile
+
+		// allow the 'create' card to appear if both lists are empty. see css.
+		var noProjectsClass = (data.publicGraphs.length + data.privateGraphs.length > 0) ? '' : 'noProjects '
+		data.meta.bodyclass = ('bUserpage ' + noProjectsClass + data.bodyclass).trim()
+		delete data.bodyclass
+
+		if (req.xhr) {
+			return res.status(200).json(
+				helper.responseStatusSuccess('OK', data))
+		}
+
+		res.render('server/pages/userpage', data)
+	}
+
+	var profile = user.toJSON()
+
+	var maxNumOnFront = 8
+	var data = {
+		publicHasMoreLink : false,
+		privateHasMoreLink : false
+	}
+	// front page, two lists
+	if (wantPrivate === undefined) {
+		data.isSummaryPage = true
+		data.bodyclass = 'bGraphlistSummary'
+		var publicList, privateList
+		that._service.userGraphs(username, {private:false}, 0, maxNumOnFront+1)
```

Comment:
@gmarinov yeah, my issue is with passing objects as parameters, because they are so unexplicit. it's an anti-pattern. can you change this to be two or three new methods instead? 

as usual, we are not programming for the future, and if we need new functionality later, we'll add it then, not code for it now.